linera-views/src/common.rs

```diff
@@ -505,21 +492,49 @@ pub trait LocalAdminKeyValueStore: Sized {
 }
 
 /// Low-level, asynchronous write and read key-value operations. Useful for storage APIs not based on views.
-pub trait KeyValueStore:
+pub trait RestrictedKeyValueStore:
     ReadableKeyValueStore<Self::Error> + WritableKeyValueStore<Self::Error>
 {
     /// The error type.
     type Error: Debug;
 }
 
 /// Low-level, asynchronous write and read key-value operations, without a `Send` bound. Useful for storage APIs not based on views.
-pub trait LocalKeyValueStore:
+pub trait LocalRestrictedKeyValueStore:
     LocalReadableKeyValueStore<Self::Error> + LocalWritableKeyValueStore<Self::Error>
 {
     /// The error type.
     type Error: Debug;
 }
 
+impl<S: RestrictedKeyValueStore> LocalRestrictedKeyValueStore for S {
+    type Error = <Self as RestrictedKeyValueStore>::Error;
+}
+
+impl<S: KeyValueStore> RestrictedKeyValueStore for S {
+    type Error = <Self as KeyValueStore>::Error;
+}
+
+/// Low-level, asynchronous write and read key-value operations. Useful for storage APIs not based on views.
+pub trait KeyValueStore:
+    ReadableKeyValueStore<Self::Error>
+    + WritableKeyValueStore<Self::Error>
+    + AdminKeyValueStore<Self::Error>
+{
+    /// The error type.
+    type Error: Debug;
+}
+
+/// Low-level, asynchronous write and read key-value operations, without a `Send` bound. Useful for storage APIs not based on views.
+pub trait LocalKeyValueStore:
+    LocalReadableKeyValueStore<Self::Error>
+    + LocalWritableKeyValueStore<Self::Error>
+    + LocalAdminKeyValueStore<Self::Error>
```

Comment:
Where is the `Send` bound? Do we really need separate traits for this or could we just add `+ Send` where needed?

Reply:
I think this is exactly what the `trait_variant::make` is doing. See https://docs.rs/trait-variant/latest/trait_variant/attr.make.html